index.js

```diff
@@ -0,0 +1,228 @@
+            this.#updateDirection(event);
+        })
+    }
+
+    #start() {
+        let middleCell = Math.floor(this.gridCount) /2;
+        this.#snake = this.#buildSnake(middleCell, middleCell, this.#size) ; 
+        this.#speed = +this.#form.speed.value; 
+         
+
+        this.#startBtn.style.display = 'none'; 
+        this.#endBtn.style.display = 'block'; 
+        this.#messageContainer.innerHTML = 'Welcome to Snake !' 
+
+        this.#process = setInterval(() => {
+
+            let { cell, row } =  this.#snake[0]; 
+            let snakePatrToShift = null; 
+
+            switch(this.direction) {
+                case D.LEFT : {
+                    snakePatrToShift = {
+                        cell: cell - 1,
+                        row
+                    }
+                }; break;
+                case D.RIGHT: {
+                    snakePatrToShift = {
+                        cell: cell + 1,
+                        row
+                    }
+                }; break;
+                case D.UP: {
+                    snakePatrToShift = {
+                        cell,
+                        row: row - 1
+                    }
+                }; break;
+                case D.DOWN: {
+                    snakePatrToShift = {
+                        cell,
+                        row: row + 1
+                    }
+                }; break;
+            };
+            snakePatrToShift = this.#noWallMode(snakePatrToShift) 
+            this.#snake.unshift(snakePatrToShift);
+            this.#clear();
+            this.#update(); 
+        },this.#speed)
+        this.#generatorFood();
+         
+ 
+    } 
+    #noWallMode({cell, row}) {
+        if (cell === -1  ) {
+            cell = cell + this.gridCount;   
+        }
+        else if (cell === 12 ) {
+            cell = 0;
+        }
+        if (row === -1  ) {
+            row = row + this.gridCount;   
+        }
+        else if (row === 12 ) {
+            row = 0;
+        } 
+        return {cell, row}
+    }
+
+    #removeАppleрEnd({pointColumn, pointRow}) {
+        let imgBlok = this.find(`[data-cell="${pointColumn}"][data-row="${pointRow}"]`, this.gridContainer);
+        imgBlok.removeChild(imgBlok.firstElementChild);
+    }
+
+    #generatorFood() {  
+        let img = document.createElement('img'); 
+        img.src = './img/apple.png';
+
+        while (true) {
+            let pointRow = this.#getRandomInt(this.gridCount);
+            let pointColumn = this.#getRandomInt(this.gridCount);
+            this.#apple = {pointColumn, pointRow};
+            let imgBlok = this.find(`[data-cell="${pointColumn}"][data-row="${pointRow}"]`, this.gridContainer);
+              
+            if (imgBlok.classList.contains('snake')) {  
+                continue;
+                console.log('====', imgBlok);
+            }
+            else {
+                imgBlok.append(img);
+                break;
+            }
+        }   
+    }
```

Comment:
`while(true)` - it's a bad code. Don't use it.
You must implement a check here so that the food does not appear in the body of the snake.
remove console.logs from the final code